src/barnetilsyn/steg/6-barnepass/Barnepass.tsx

```diff
@@ -81,38 +90,59 @@ const Barnepass: FC<Props> = () => {
     settDokumentasjonsbehov(spørsmål, svar);
   };
 
-  const erBarnepassSpmBesvart: boolean =
-    (søkerFraBestemtMåned?.svarid === ESøkerFraBestemtMåned.ja &&
-      søknadsdato?.verdi !== undefined) ||
-    søkerFraBestemtMåned?.svarid === ESøkerFraBestemtMåned.neiNavKanVurdere;
-
   return (
     <Side
       tittel={intl.formatMessage({ id: 'barnepass.sidetittel' })}
       skalViseKnapper={true}
       mellomlagreBarnetilsyn={mellomlagreBarnetilsyn}
-      erSpørsmålBesvart={erBarnepassSpmBesvart}
+      erSpørsmålBesvart={erBarnepassStegFerdigUtfylt(
+        barnSomSkalHaBarnepass,
+        søknad
+      )}
     >
-      {barnSomSkalHaBarnepass.map((barn) => (
-        <SeksjonGruppe key={barn.id}>
-          <BarnepassOrdninger
-            barn={barn}
-            settBarnepass={settBarnepass}
-            settDokumentasjonsbehov={settDokumentasjonsbehov}
-          />
-        </SeksjonGruppe>
-      ))}
       <SeksjonGruppe>
-        <NårSøkerDuStønadFra
-          spørsmål={SøkerDuStønadFraBestemtMndSpm}
-          settSøkerFraBestemtMåned={settSøkerFraBestemtMåned}
-          settDato={settSøknadsdato}
-          søkerFraBestemtMåned={søkerFraBestemtMåned}
-          valgtDato={søknadsdato}
-          datovelgerLabel={datovelgerLabel}
-          hjelpetekstInnholdTekstid={hjelpetekstInnholdTekstid}
-        />
+        {barnSomSkalHaBarnepass.map((barn, index) => {
+          const visSeksjon =
+            index === 0 ||
+            erBarnepassForBarnFørNåværendeUtfylt(barn, barnSomSkalHaBarnepass);
+          return (
+            visSeksjon && (
+              <>
```

Comment:
Denne må nok endres til `<React.Fragment key={id}>` for at `React` skal være hyggelig med oss

Reply:
`id` kan være `barn.id` her

Reply:
✅